cmake/macros/SundialsAddBenchmark.cmake

```diff
@@ -29,62 +29,57 @@ macro(sundials_add_benchmark NAME EXECUTABLE BASE_BENCHMARK_NAME)
   # ENABLE_GPU        = indicate this benchmark should be run with GPUs
   set(options ENABLE_GPU)
 
-  cmake_parse_arguments(sundials_add_benchmark "${options}" "${oneValueArgs}"
-                        "${multiValueArgs}" ${ARGN})
+  cmake_parse_arguments(arg "${options}" "${oneValueArgs}" "${multiValueArgs}"
```

Comment:
Is it possible that this could cause variable name conflict, e.g., if you call another macro from within this one that has an argument of the same name?

Reply:
Yes it could. I've switched this and `sundials_add_test` from a `macro` to a `function`. Side note, I got this idea to simplify the prefix looking at some BLT macros and I suspect they might run into the same scoping issues.